swf/src/read.rs

```diff
@@ -2427,15 +2432,35 @@ impl<R: Read> Reader<R> {
         })
     }
 
-    pub fn read_define_text(&mut self) -> Result<Text> {
+    pub fn read_define_text_1(&mut self) -> Result<Text> {
+        let id = self.read_character_id()?;
+        let bounds = self.read_rectangle()?;
+        let matrix = self.read_matrix()?;
+        let num_glyph_bits = self.read_u8()?;
+        let num_advance_bits = self.read_u8()?;
+
+        let mut records = vec![];
+        while let Some(record) = self.read_text_record_1(num_glyph_bits, num_advance_bits)? {
+            records.push(record);
+        }
+
+        Ok(Text {
+            id,
+            bounds,
+            matrix,
+            records,
+        })
+    }
+
+    pub fn read_define_text_2(&mut self) -> Result<Text> {
```

Comment:
Instead of duplicating most of this code, let's pass `version` of 1 or 2 down as a parameter and use that in `read_text_record` to decide whether to read RGB or RGBA

Reply:
:+1: makes sense. I did originally do that but somehow I missed the examples of where this is done so assumed that this pattern was used for whatever reason. Must have had my eyes closed though as now I see it everywhere. Thanks, much nicer solution :)

Reply:
I'll do separate functions if the tag versions are different enough (`read_define_font` is an example), although even I've been inconsistent with it sometimes!